Unable to handle - IllegalStateException in Jetty Client Request send()

**Jetty version(s)**
12.0.13

**Jetty Environment**
ee10

**Java version/vendor** `(use: java -version)`
Java 21

**OS type/version**
Linux (Oracle Linux 8)

**Description**
We encountered an issue with the request.send() call which throws an IllegalStateException: ContentSender[ABORTED]

Expected Behavior
According to the Jetty documentation, send() should only throw:
    InterruptedException (if the thread is interrupted)
    TimeoutException (if the request times out)
    ExecutionException (if execution fails)

However, we are seeing the following error:

```
Caused by: java.lang.IllegalStateException: ContentSender@e9f4ba3[ABORTED]
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:239)
	at org.eclipse.jetty.client.transport.HttpSender.send(HttpSender.java:85)
	at org.eclipse.jetty.client.transport.internal.HttpChannelOverHTTP.send(HttpChannelOverHTTP.java:86)
	at org.eclipse.jetty.client.transport.HttpChannel.send(HttpChannel.java:142)
	at org.eclipse.jetty.client.transport.HttpConnection.send(HttpConnection.java:112)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:379)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:168)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:444)
	at org.eclipse.jetty.client.transport.HttpDestination.process(HttpDestination.java:420)
	at org.eclipse.jetty.client.transport.HttpDestination.process(HttpDestination.java:375)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:358)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:352)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:329)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:308)
	at org.eclipse.jetty.client.transport.HttpRequest.sendAsync(HttpRequest.java:751)
	at org.eclipse.jetty.client.transport.HttpDestination.send(HttpDestination.java:303)
	at org.eclipse.jetty.client.transport.HttpRequest.send(HttpRequest.java:744)
	at org.eclipse.jetty.client.CompletableResponseListener.send(CompletableResponseListener.java:79)
	at org.eclipse.jetty.client.transport.HttpRequest.send(HttpRequest.java:707)
```

**How to reproduce?**
1. Send a request using send() to a service that does not respond within the timeout period.
2. Observed that sometimes, instead of throwing TimeoutException, Jetty throws IllegalStateException: ContentSender[ABORTED].

--- a/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java
+++ b/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java
@@ -215,11 +215,6 @@ public abstract class IteratingCallback implements Callback
         {
             switch (_state)
             {
-                case PENDING:
-                case CALLED:
-                    // process will be called when callback is handled
-                    break;
-
                 case IDLE:
                     _state = State.PROCESSING;
                     process = true;
@@ -229,14 +224,8 @@ public abstract class IteratingCallback implements Callback
                     _iterate = true;
                     break;
 
-                case FAILED:
-                case SUCCEEDED:
-                    break;
-
-                case CLOSED:
-                case ABORTED:
                 default:
-                    throw new IllegalStateException(toString());
+                    break;
             }
         }
         if (process)

